### DISCUSSION TITLE
Custom nextjs root folder?

### DISCUSSION THREAD

**Discussion by lanyusan** [vercel/next.js#22620] [Help]
Is there are way to set a custom root folder?

For some reason, I need to put everything of a nextjs project for example pages, src, etc. to another folder. Is there a way to do so, for example set an environment variable like NEXJS_ROOT etc.?

I understand it is an unusual requirement but no need to discuss why here as it is NOT this question for.

Thanks.

**Comment by michrome** [comment]
If you can't use `/pages` then an alternative option is **[`/src/pages`](https://nextjs.org/docs/advanced-features/src-directory)** (note `/pages` must not exist).
